_posts/pl/2021-01-11-rest-api-dobre-praktyki.md

```diff
@@ -112,12 +209,34 @@ Response: 404
+Metoda PATCH została wprowadzona do standardu w RFC 5789 w 2010 roku, nie był to jednak wówczas nowy pomysł. Koncepcja opiera się na tym, że w przeciwieństwie do metody PUT, nie przekażemy całego zasobu, a jedynie zmiany. W związku z tym metoda nie jest idempotentna w przeciwieństwie do metody PUT, to znaczy, że metoda PUT wywołana dowolną liczbę razy da w efekcie zawsze taki sam stan zasobu. W przypadku PATCH-a nie zawsze tak będzie.
 
 // TODO przykład PUT i PATCH
 
-Chcąc więc być spójni ze standardem i stosując tę metodę HTTP w odpowiednim przypadku, mamy poprawne rozwiązanie, przetestowane na środowisku deweloperskim. Dopiero na kolejnym etapie okazuje się, że żądania PATCH trafiające na nasz serwer są po drodze wycinane - blokowane przez WAF. WAF, którego celem jest zapewnienie bezpieczeństwa, jest konfigurowany zazwyczaj przez inny zespół, czasem z innej firmy lub przez klienta, dla którego dostarczamy rozwiązanie. Dlatego tego typu problemy są dosyć częste. Konfiguracja WAF, oparta na blokowaniu niedozwolonych żądań i odpowiedzi lub akceptowaniu tylko dozwolonych, to niezwykle istotny element bezpieczeństwa.
+Chcąc więc być spójni ze standardem i stosując tę metodę HTTP w odpowiednim przypadku, mamy poprawne rozwiązanie, przetestowane na środowisku deweloperskim. Dopiero na kolejnym etapie okazuje się, że żądania PATCH trafiające na nasz serwer są po drodze wycinane — blokowane przez WAF. WAF, którego celem jest zapewnienie bezpieczeństwa, jest konfigurowany zazwyczaj przez inny zespół, czasem z innej firmy lub przez klienta, dla którego dostarczamy rozwiązanie. Dlatego tego typu problemy są dosyć częste. Konfiguracja WAF jest oparta na blokowaniu niedozwolonych żądań i odpowiedzi lub akceptowaniu tylko dozwolonych.
+
+Czy w opisanym przypadku blokowanie żądań, które korzystają z metody PATCH, było uzasadnione? Zapewne wynikało z tego, że konfiguracja była zawężona najbardziej, jak się dało. W tym przypadku można oczywiście zmienić konfigurację WAF, ale nierzadko w tego typu sytuacjach w praktyce dostosowujemy się do infrastruktury klienta, któremu dostarczamy oprogramowanie i decydujemy się na zmianę np. na PUT. W ten sposób nie jesteśmy już zgodni ze standardem. Widać tu wyraźnie różnicę między sytuacją, gdy wystawiamy API dla potencjalnie wielu klientów i trzymanie się standardów ułatwi innym zrozumienie naszego API a sytuacją, gdy tworzymy rozwiązanie dla jednego klienta, które działa na jego infrastrukturze i staramy się być elastyczni. W drugim przypadku jakość naszego API będzie dużo niższa. Nie można jednak zapominać, że zazwyczaj reguły na WAFie służą poprawie bezpieczeństwa i zabezpieczeniem przed znanymi i dobrze opisanymi atakami. Dlatego często to właśnie brak trzymania się standardów doprowadzi do problemów na WAF-ie.
```

Comment:
Chociaż zdanie jest jasne, to wpuściłbym tutaj trochę powietrza:

Widać tu wyraźnie różnicę między sytuacją, gdy wystawiamy API dla potencjalnie wielu klientów i wówczas trzymanie się standardów ułatwi innym zrozumienie naszego API, a sytuacją, gdy tworzymy rozwiązanie dla jednego klienta, które działa na jego infrastrukturze, a my staramy się być elastyczni.